data-prepper-plugins/opensearch/src/main/java/com/amazon/dataprepper/plugins/sink/opensearch/OpenSearchSink.java

```diff
@@ -142,6 +143,16 @@ public void doOutput(final Collection<Record<String>> records) {
     }
   }
 
+  private String getDocument(final Object object) {
+    if (object instanceof String) {
+      return (String) object;
```

Comment:
Following the above comment, I do not think a generic string can be exported by index request. Especially in L122 the source type is referred as XContentType.JSON

Reply:
This code path supports the original functionalty. This isnt just a String, it is a Json String. If you notice the original L120 was mapping `record.Data()` to the document. The new L121 does the exact same thing except leverages `getDocument` to ensure the type is already a string or converts the Event to a Json String. All trace analytics preppers pass Records with son Strings.

This method is only temporary and should be eliminated with #546  is completed and trace plugins are migrated to the event model.